waves/project.py

```diff
@@ -2846,3 +2892,404 @@ def generate_report_lcoe_breakdown(self) -> pd.DataFrame:
+                ),  # Average annual wind speed at hub height
+                np.mean(
+                    compute_shear(
+                        self.wombat.env.weather.to_pandas(),
+                        self.identify_windspeed_columns_and_heights(
+                            self.wombat.env.weather.to_pandas()
+                        ),
+                        False,
+                    )
+                ),  # Shear exponent
+                self.compute_weibull(self.orbit.config["turbine"]["hub_height"]),  # Weibull k
+                100 * self.loss_ratio(),  # Total system losses
+                100
+                * self.availability(
+                    which="energy", frequency="project", by="windfarm"
+                ),  # Availability
+                self.energy_potential(
+                    units="mw", per_capacity="mw", aep=True
+                ),  # Gross energy capture
+                self.energy_production(
+                    units="mw", per_capacity="mw", aep=True
+                ),  # Net energy capture
+                100
+                * self.capacity_factor(
+                    which="gross", frequency="project", by="windfarm"
+                ),  # Gross capacity factor
+                100
+                * self.capacity_factor(
+                    which="net", frequency="project", by="windfarm"
+                ),  # Net capacity factor
+            ],
+        }
+
+        # Create a DataFrame from the template
+        df = pd.DataFrame(project_details)
+
+        return df
+
+    def determine_substructure_type(self):
+        """Determine the substructure type based on the ORBIT configuration file.
+
+        This function scans the "design_phases" section of the ORBIT configuration file to identify
+        the substructure type used in the project. The substructure types considered are
+        "Monopile", "SemiSubmersible", "Jacket", and "Spar". The function returns the substructure
+        type as a string if found in the design phases, or "Unknown" if no substructure type is
+        identified.
+
+        Returns
+        -------
+        str
+            The substructure type as one of the following: "Monopile", "SemiSubmersible",
+            "Jacket", "Spar", or "Unknown" if no match is found.
+
+        Notes
+        -----
+        - The search is case-insensitive and looks for the substructure types as substrings within
+          the design phase names.
+        - If no substructure type is found after checking all phases, the function will return
+          "Unknown".
+        """
+        if self.turbine_type == "land":
+            raise NotImplementedError("Land-based analyses are not yet supported.")
+
+        project_data = self.orbit.config["design_phases"]
+
+        # Iterate over the design_phases to check for the substructure type
+        if "MonopileDesign" in project_data:
+            return "Monopile"
+        elif "SemiSubmersibleDesign" in project_data:
```

Comment:
I forgot to mention this before, but the `elif` statements can all be separate `if` statements, which simplifies the control flow. Additionally, the `else:` line should be removed, and the return statement should be dedented so that the method always returns one of the expected values. This last piece is important because in its current form, the method has the ability to silently fail (return a `None` if none of the design phases are caught).

As an aside, when we bump the minimum Python version to 3.10+, this will be really nice as a match-case statement.